ToolBox/SECDataDownloader/SECDataConverter.cs

```diff
@@ -0,0 +1,361 @@
+                        {
+                            // Avoid processing files greater than MaxFileSize megabytes
+                            if (MaxFileSize < new FileInfo(rawReportFilePath).Length)
+                            {
+                                Log.Trace($"SECDataConverter.Process(): File {rawReportFilePath} is too large to process. Continuing...");
+                                return;
+                            }
+
+                            var factory = new SECReportFactory();
+                            var xmlText = new StringBuilder();
+
+                            // We need to escape any nested XML to ensure our deserialization happens smoothly
+                            var parsingText = false;
+
+                            foreach (var line in File.ReadLines(rawReportFilePath))
+                            {
+                                var newTextLine = line;
+                                var currentTagName = factory.GetTagNameFromLine(newTextLine);
+
+                                // This tag is present rarely in SEC reports, but is unclosed without value when encountered.
+                                // Verified by searching with ripgrep for "CONFIRMING-COPY"
+                                if (currentTagName == "CONFIRMING-COPY")
+                                {
+                                    return;
+                                }
+
+                                // Don't encode the closing tag
+                                if (currentTagName == "/TEXT")
+                                {
+                                    parsingText = false;
+                                }
+
+                                // To ensure that we can serialize/deserialize data with hours, minutes, seconds
+                                if (currentTagName == "FILING-DATE" || currentTagName == "PERIOD" ||
+                                    currentTagName == "DATE-OF-FILING-CHANGE" || currentTagName == "DATE-CHANGED")
+                                {
+                                    newTextLine = $"{newTextLine.TrimEnd()} 00:00:00";
+                                }
+
+                                // Encode all contents inside tags to prevent errors in XML parsing.
+                                // The json deserializer will convert these values back to their original form
+                                if (!parsingText && factory.HasValue(newTextLine))
+                                {
+                                    newTextLine =
+                                        $"<{currentTagName}>{SecurityElement.Escape(factory.GetTagValueFromLine(newTextLine))}</{currentTagName}>";
+                                }
+                                // Escape all contents inside TEXT tags
+                                else if (parsingText)
+                                {
+                                    newTextLine = SecurityElement.Escape(newTextLine);
```

Comment:
shame all this pre-processing cleaning was needed, wouldn't the world be great if we always had clean data :D

Reply:
At least it's very consistent and no big gotchas existed 😄